### PR TITLE
Course annotation tool

### PR DESCRIPTION
**Pull request by pkwagner**
In addition to the already implemented course hiding functionality, this PR allows choosing an individual annotation color for each module.

Screenshot 2020-10-06 at 16 51 04

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -27,13 +27,19 @@ <h2>TUD Informatik Bachelor VV {{semester}}</h2>
...
+                <div data-angular-treeview="true" data-tree-id="tree" data-tree-model="data" data-node-id="label" data-node-label="label" data-node-children="children" data-label-template="<span class='annotation-((node.annotation))'>((node.label)) (( labelForNode(node) ))</span><span class='credits no-wrap credits-((node.credits))' ng-if='node.credits'>((node.credits)) CP</span>">
                 </div>
             </div>
         </section>
         <aside ng-if="selectedNode" scroll-top-on-change="selectedNode" ng-cloak>
             <div class="close-button button round-button" ng-click="showSelected()">x</div>
             <button ng-click="toggleNodeHiddenStatus(selectedNode, true); showSelected()">Diesen Kurs nicht mehr anzeigen</button>
+
+            <button ng-click="setAnnotation(selectedNode, 1)" class="annotation annotation-1" ng-class="{'annotation-active': selectedNode.annotation === 1}">&nbsp;</button>
```
Comment (davidgengenbach) [inline index.html]: The following lines are a little repetitious - can you maybe put the annotations (in this case `[1, 2, 3, 4, 5]`) in the `$scope` and repeat via `ng-repeat` over them (= generating the buttons dynamically).

File: index.html [hunk 2]
```diff
@@ -27,13 +27,19 @@ <h2>TUD Informatik Bachelor VV {{semester}}</h2>
                 <button class="button-show-hidden-courses" ng-if="hiddenCourses().length > 0" class="border-left" ng-click="config.showHiddenCoursesDialog = true">
                     {{hiddenCourses().length}} ausgeblendete Kurse anzeigen
                 </button>
-                <div data-angular-treeview="true" data-tree-id="tree" data-tree-model="data" data-node-id="label" data-node-label="label" data-node-children="children" data-label-template="((node.label)) (( labelForNode(node) ))<span class='credits no-wrap credits-((node.credits))' ng-if='node.credits'>((node.credits)) CP</span>">
+                <div data-angular-treeview="true" data-tree-id="tree" data-tree-model="data" data-node-id="label" data-node-label="label" data-node-children="children" data-label-template="<span class='annotation-((node.annotation))'>((node.label)) (( labelForNode(node) ))</span><span class='credits no-wrap credits-((node.credits))' ng-if='node.credits'>((node.credits)) CP</span>">
```
Comment (davidgengenbach) [inline index.html]: I know this is more of my fault but can you please break this rather long line into more lines?

**Review by davidgengenbach — CHANGES_REQUESTED** [review]

**Comment by davidgengenbach** [comment]
Very nice addition!

I will not maintain this project any longer as I finished my studies at TU Darmstadt.
Would you be interested in becoming the maintainer of this repository (and maybe the whole TuCan "suite", e.g., `tucan-tools`, as well)?

**Comment by pkwagner** [comment]
Thanks for the feedback, I'll push it later today / tomorrow!

**Review by davidgengenbach — APPROVED** [review]

**Pull request merged**